smarts/core/tests/test_map.py

```diff
@@ -127,3 +138,530 @@ def test_sumo_map(scenario):
             assert any(
                 r2lane == cand[0] and math.isclose(cand[1], 53.6) for cand in cands
             )
+
+
+def test_opendrive_map_4lane(opendrive_scenario_4lane):
+    road_map = opendrive_scenario_4lane.road_map
+    assert isinstance(road_map, OpenDriveRoadNetwork)
+
+    # Expected properties for all roads and lanes
+    for road_id, road in road_map._roads.items():
+        assert type(road_id) == str
+        assert road.is_junction is not None
+        assert road.length is not None
+        assert road.length >= 0
+        assert road.parallel_roads == []
```

Comment:
there is no way we can find `parallel_roads` in `OpenDriveRoadNetwork` since they depend on `nodes` between roads that are exclusive to `sumolib.net`. So `parallel_roads` technically don't exist for OpenDRIVE and hence we return empty list here if called.

Reply:
I don't think we can do that though. `parallel_roads` is used in `smarts/core/sensors.py`,
```python
class RoadWaypointsSensor(Sensor):
    def __init__(self, vehicle, sim, plan, horizon=32):
        self._vehicle = vehicle
        self._road_map = sim.road_map
        self._plan = plan
        self._horizon = horizon

    def __call__(self):
        veh_pt = self._vehicle.pose.point
        lane = self._road_map.nearest_lane(veh_pt)
        if not lane:
            return RoadWaypoints(lanes={}, route_waypoints=[])
        road = lane.road
        lane_paths = {}
        for croad in (
            [road] + road.parallel_roads + road.oncoming_roads_at_point(veh_pt)
        ):
            for lane in croad.lanes:
                lane_paths[lane.lane_id] = self.paths_for_lane(lane)
```
So inorder to use it we need to check if `self._road_map` above is an instance of `SumoRoadNetwork`.  For us to do this we need to import `SumoRoadNetwork` object which may lead to dependency on other RoadMap objects like when`OpendriveRoadNetwork` calls the above code snippet.

Reply:
Ah good point.  I forgot about this usage of it.